frontend/src/pages/UsersPageServices/UsersPageService.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { findService } from '../../models/devices'
+import { useSelector } from 'react-redux'
+import { ApplicationState } from '../../store'
+import { Container } from '../../components/Container'
+import { useHistory } from 'react-router-dom'
+import analytics from '../../helpers/Analytics'
+import { SharedUsersList } from '../../components/SharedUsersList'
+import { SharedUsersHeader } from '../../components/SharedUsersHeader'
+
+export const UsersPageService: React.FC = () => {
+  const { serviceID = '' } = useParams()
+  const { deviceID = '' } = useParams()
+  const { device, service } = useSelector((state: ApplicationState) => ({
+    device: state.devices.find((d: IDevice) => d.id === deviceID),
+    service: findService(state.devices.all, serviceID)[0]
+  }))
+  const shared =  service?.access.length
+
+  const history = useHistory()
+
+  useEffect(() => {
+    analytics.page('UsersPageService')
+  }, [])
+
+  const onClick = () => history.push(`/devices/${deviceID}/share`)
```

Comment:
You'll want to use the location.path to build this link, as you could be coming from the connections tab route or the device tab route.
`location.pathname + '/share'`
Please check that this works... additional routes might need to be added to the router.tsx page for support through the connections tab.